packages/valory/skills/check_stop_trading_abci/behaviours.py

```diff
@@ -0,0 +1,156 @@
+            contract_public_id=MechContract.contract_id,
+            contract_callable="get_requests_count",
+            data_key="requests_count",
+            placeholder=get_name(CheckStopTradingBehaviour.mech_request_count),
+            address=self.synchronized_data.safe_contract_address,
+        )
+        return status
+
+    @property
+    def is_first_period(self) -> bool:
+        """Return whether it is the first period of the service."""
+        return self.synchronized_data.period_count == 0
+
+    @property
+    def params(self) -> CheckStopTradingParams:
+        """Return the params."""
+        return cast(CheckStopTradingParams, self.context.params)
+
+    def is_staking_kpi_met(self) -> WaitableConditionType:
+        """Return whether the staking KPI has been met."""
+
+        yield from self.wait_for_condition_with_sleep(self._get_mech_request_count)
+        mech_request_count = self.mech_request_count
+        self.context.logger.info(f"{self.mech_request_count=}")
+
+        yield from self.wait_for_condition_with_sleep(self._get_service_info)
+        mech_request_count_on_last_checkpoint = self.service_info[2][1]
+        self.context.logger.info(f"{mech_request_count_on_last_checkpoint=}")
+
+        yield from self.wait_for_condition_with_sleep(self._get_ts_checkpoint)
+        last_ts_checkpoint = self.ts_checkpoint
+        self.context.logger.info(f"{last_ts_checkpoint=}")
+
+        yield from self.wait_for_condition_with_sleep(self._get_liveness_period)
+        liveness_period = self.liveness_period
+        self.context.logger.info(f"{liveness_period=}")
+
+        yield from self.wait_for_condition_with_sleep(self._get_liveness_ratio)
+        liveness_ratio = self.liveness_ratio
+        self.context.logger.info(f"{liveness_ratio=}")
+
+        mech_requests_since_last_cp = mech_request_count - mech_request_count_on_last_checkpoint
+        self.context.logger.info(f"{mech_requests_since_last_cp=}")
+
+        current_timestamp = self.synced_timestamp
+        self.context.logger.info(f"{current_timestamp=}")
+
+        required_mech_requests = math.ceil(max(
+            (current_timestamp - last_ts_checkpoint) * liveness_ratio / 10**18,
+            (liveness_period) * liveness_ratio / 10**18
+        )) + 1
+        self.context.logger.info(f"{required_mech_requests=}")
+
+        if mech_requests_since_last_cp >= required_mech_requests:
+            return True
+        return False
+
+    def async_act(self) -> Generator:
+        """Do the action."""
+        with self.context.benchmark_tool.measure(self.behaviour_id).local():
+
+            # This is a "hacky" way of getting required data initialized on
+            # the Trader: On first period, the FSM needs to initialize some
+            # data on the trading branch so that it is available in the
+            # cross-period persistent keys.
+            if self.is_first_period:
+                stop_trading = False
```

Comment:
This forces the Trader to always go to the full "bet placement" cycle at least once, when restarted. This ensures that the values in the persitent data are initialized. Of course this is is a "hacky" way. However, this was the simplest way to address this issue. An alternative to fix this will be prepared in a different PR, in order to avoid too much refactoring of existing components in this PR.

Reply:
Let's open an issue for that and not release until this is done.

Reply:
Opened an issue https://github.com/valory-xyz/trader/issues/246

Reply:
I don't think we should block the release for this issue. Of course it's not the ideal behaviour (it always require one mech call at restart) but at least it's better than the current implementation (it requires indefinite amount of mech calls).

Reply:
Ok, but this needs to be clear in the release announcement.

Reply:
Ok, it will be included in the release announcement.